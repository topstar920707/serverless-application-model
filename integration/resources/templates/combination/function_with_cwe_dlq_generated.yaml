Resources:
  MyLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs14.x
      CodeUri: ${codeuri}
      MemorySize: 128

      Events:
        CWEvent:
          Type: CloudWatchEvent
          Properties:
            Pattern:
              detail:
                state:
                - terminated
            RetryPolicy:
              MaximumRetryAttempts: 6
              MaximumEventAgeInSeconds: 900
            DeadLetterConfig:
              Type: SQS
              QueueLogicalId: MyDlq

Outputs:
  MyLambdaArn:
    Description: "Arn of the Lambda target"
    Value:
      Fn::GetAtt:
        - "MyLambdaFunction"
        - "Arn"
  MyEventName:
    Description: "Name of the CWE rule created"
    Value:
      Ref: MyLambdaFunctionCWEvent
  MyDLQArn:
    Description: "Arn of the dead-letter queue provided for the CWE rule target"
    Value:
      Fn::GetAtt:
        - "MyDlq"
        - "Arn"
  MyDLQUrl:
    Description: "Url of the dead-letter queue provided for the CWE rule target"
    Value:
      Ref: MyDlq

