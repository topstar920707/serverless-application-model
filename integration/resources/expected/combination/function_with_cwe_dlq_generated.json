[
  { "LogicalResourceId":"MyLambdaFunction", "ResourceType":"AWS::Lambda::Function" },
  { "LogicalResourceId":"MyLambdaFunctionRole", "ResourceType":"AWS::IAM::Role" },
  { "LogicalResourceId":"MyLambdaFunctionCWEvent", "ResourceType":"AWS::Events::Rule" },
  { "LogicalResourceId":"MyLambdaFunctionCWEventPermission", "ResourceType":"AWS::Lambda::Permission" },
  { "LogicalResourceId":"MyDlq", "ResourceType":"AWS::SQS::Queue" },
  { "LogicalResourceId":"MyLambdaFunctionCWEventQueuePolicy", "ResourceType":"AWS::SQS::QueuePolicy" }
]